exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +81,459 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+					UID:        ownerMachinePool(i).UID,
+					Kind:       "MachinePool",
+					APIVersion: clusterv1.GroupVersion.String(),
+				},
+			}
+			cp := rosaControlPlane(i)
+			objects := []client.Object{ownerCluster(i), ownerMachinePool(i), cp, test.old}
+
+			for _, obj := range objects {
+				createObject(g, obj, ns.Name)
+			}
+			// make Control Plane ready, can't do this duirng creation
+			mpPh, err := patch.NewHelper(cp, testEnv)
+			cp.Status.Ready = true
+			g.Expect(mpPh.Patch(ctx, cp)).To(Succeed())
+			g.Expect(err).ShouldNot(HaveOccurred())
+			// patching is not reliably synchronous
+			time.Sleep(50 * time.Millisecond)
+
+			mockCtrl := gomock.NewController(t)
+			recorder := record.NewFakeRecorder(10)
+			ctx := context.TODO()
+			ocmMock := mocks.NewMockOCMClient(mockCtrl)
+			test.expect(ocmMock.EXPECT())
+
+			stsMock := mock_stsiface.NewMockSTSAPI(mockCtrl)
+			stsMock.EXPECT().GetCallerIdentity(gomock.Any()).Times(1)
+
+			r := ROSAMachinePoolReconciler{
+				Recorder:         recorder,
+				WatchFilterValue: "",
+				Endpoints:        []scope.ServiceEndpoint{},
+				Client:           testEnv,
+				NewStsClient:     func(cloud.ScopeUsage, cloud.Session, logger.Wrapper, runtime.Object) stsiface.STSAPI { return stsMock },
+				NewOCMClient: func(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (rosa.OCMClient, error) {
+					return ocmMock, nil
+				},
+			}
+
+			req := ctrl.Request{}
+			req.NamespacedName = types.NamespacedName{Name: test.old.Name, Namespace: ns.Name}
+
+			result, errReconcile := r.Reconcile(ctx, req)
+			g.Expect(errReconcile).ToNot(HaveOccurred())
+			g.Expect(result).To(Equal(test.result))
+			time.Sleep(50 * time.Millisecond)
+
+			m := &expinfrav1.ROSAMachinePool{}
+			key := client.ObjectKey{Name: test.old.Name, Namespace: ns.Name}
+			errGet := testEnv.Get(ctx, key, m)
+			g.Expect(errGet).NotTo(HaveOccurred())
+			g.Expect(m.Status.Ready).To(Equal(test.new.Status.Ready))
+			g.Expect(m.Status.Replicas).To(Equal(test.new.Status.Replicas))
+			g.Expect(m.Status.ID).To(Equal(test.new.Status.ID))
+
+			// cleanup
+			for _, obj := range objects {
+				cleanupObject(g, obj)
+			}
+			mockCtrl.Finish()
+		})
+	}
+
+	t.Run("Reconcile delete", func(t *testing.T) {
+		g := NewWithT(t)
+		mockCtrl := gomock.NewController(t)
+		recorder := record.NewFakeRecorder(10)
+		ctx := context.TODO()
+		mp := &expinfrav1.ROSAMachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "rosa-machinepool-9",
+				Namespace:  ns.Name,
+				UID:        "rosa-machinepool-9",
+				Finalizers: []string{expinfrav1.RosaMachinePoolFinalizer},
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ROSAMachinePool",
+				APIVersion: expinfrav1.GroupVersion.String(),
+			},
+			Spec: expinfrav1.RosaMachinePoolSpec{
+				NodePoolName: "test-nodepool-1",
+				Version:      "4.14.5",
+
+				Subnet:       "subnet-id",
+				InstanceType: "m5.large",
+			},
+		}
+		oc := ownerCluster(9)
```

Comment:
So here you are passing 9 to be similar to the rosaMachinePool naming, its better to use the rand lib generate random string and pass it to all the test CRs